dpnp/backend/kernels/dpnp_krnl_indexing.cpp

```diff
@@ -1074,15 +1072,23 @@ void func_map_init_indexing_func(func_map_t& fmap)
                                                                         (void*)dpnp_put_along_axis_ext_c<double>};
 
     fmap[DPNPFuncName::DPNP_FN_TAKE][eft_BLN][eft_BLN] = {eft_BLN, (void*)dpnp_take_default_c<bool, int64_t>};
-    fmap[DPNPFuncName::DPNP_FN_TAKE][eft_INT][eft_INT] = {eft_INT, (void*)dpnp_take_default_c<int32_t, int64_t>};
+    fmap[DPNPFuncName::DPNP_FN_TAKE][eft_INT][eft_INT] = {eft_INT, (void*)dpnp_take_default_c<int32_t, int32_t>};
```

Comment:
Clearly these are not all combinations. What's happening in `dpnp` in case when the type combination is not found in `fmap`? 

I also think that `[eft_C128][eft_C128]` was really intended for the use-case where function pointer lookup depended only on the source array type, and not on the type of indices.

Reply:
> Clearly these are not all combinations. What's happening in `dpnp` in case when the type combination is not found in `fmap`?

It will result in runtime exception, like:
`RuntimeError: DPNP Error: Function ID with unsupported second parameter type.`

> I also think that `[eft_C128][eft_C128]` was really intended for the use-case where function pointer lookup depended only on the source array type, and not on the type of indices.

Yep, agree. In case of numpy there will be a type error, like:
`TypeError: Cannot cast array data from dtype('complex256') to dtype('int64') according to the rule 'safe'`